hal/include/HAL/Types.h

```diff
@@ -44,3 +44,13 @@ typedef HAL_Handle HAL_RelayHandle;
 typedef HAL_Handle HAL_SolenoidHandle;
 
 typedef int32_t HAL_Bool;
+
+#ifdef __cplusplus
+#define HAL_ENUM_START(name, type) enum name : type
+#define HAL_ENUM_END(name, type) ; /* NOLINT */
+#else
+#define HAL_ENUM_START(name, type) enum
+#define HAL_ENUM_END(name, type) \
+  ;                              \
+  typedef type name;
```

Comment:
In the c++ case the start block end triplet evaluate to:

    enum MyEnumType : int32_t { VALUES... };

In C:

    enum { VALUES... }; typedef int32_t MyEnumType;

If the C++ code didn't use `MyEnumType::Value`, then this would be much simpler.

I'll stick some documentation on it.